pkg/graphviz/traceflow.go

```diff
@@ -115,9 +115,31 @@ func getDstNodeName(tf *opsv1alpha1.Traceflow) string {
 	if len(tf.Spec.Destination.Namespace) > 0 && len(tf.Spec.Destination.Pod) > 0 {
 		return tf.Spec.Destination.Namespace + "/" + tf.Spec.Destination.Pod
 	}
+	if len(tf.Spec.Destination.Namespace) > 0 && len(tf.Spec.Destination.Service) > 0 {
+		return tf.Spec.Destination.Namespace + "/" + tf.Spec.Destination.Service
+	}
+	if len(tf.Spec.Destination.IP) > 0 {
+		return tf.Spec.Destination.IP
+	}
 	return ""
 }
 
+// GetTraceflowMessage gets the shown message string in traceflow graph.
+func GetTraceflowMessage(o opsv1alpha1.Observation) string {
```

Comment:
Does it need to be public?

Reply:
Use pointer to avoid struct copy given it has ten more fields.

Reply:
Changed GetTraceflowMessage and other 2 functions to private functions and use pointers.